arrow-core-data/src/main/kotlin/arrow/core/Eval.kt

```diff
@@ -244,15 +243,18 @@ sealed class Eval<out A> : EvalOf<A> {
 
   abstract fun memoize(): Eval<A>
 
-  fun <B> map(f: (A) -> B): Eval<B> = flatMap { a -> Now(f(a)) }
+  inline fun <B> map(crossinline f: (A) -> B): Eval<B> =
+    flatMap { a -> Now(f(a)) }
 
-  fun <B> ap(ff: EvalOf<(A) -> B>): Eval<B> = ff.fix().flatMap { f -> map(f) }.fix()
+  fun <B> ap(ff: EvalOf<(A) -> B>): Eval<B> =
+    ff.fix().flatMap { f -> map(f) }.fix()
 
   @Suppress("PARAMETER_NAME_CHANGED_ON_OVERRIDE", "UNCHECKED_CAST")
-  fun <B> flatMap(f: (A) -> EvalOf<B>): Eval<B> =
+  inline fun <B> flatMap(crossinline f: (A) -> EvalOf<B>): Eval<B> =
     when (this) {
       is FlatMap<A> -> object : FlatMap<B>() {
         override fun <S> start(): Eval<S> = (this@Eval).start()
+        @IgnoreJRERequirement
```

Comment:
Can you clarify the need for this? 🙏

Reply:
This is caused due to a likely false positive of Animal-Sniffer not being able to process this, as soon as I added the inline it started failing here 🤷

Reply:
Hmm, could it be that Kotlinc is generating code not compatible with Android? This is the only doc I could find on that annotation, but it clearly states that it is used to mark code sites that `knowingly use features that go beyond the minimum requirement`.

https://github.com/mojohaus/animal-sniffer/blob/fdc17216a51a57a408d384e6af12830ff685cb5f/animal-sniffer/src/site/apt/signature-checker.apt#L123